src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.mapvalidator.KeyPresentValidator;
+import java.util.Map;
+
+public class PurchaseOperationHandler
+        extends AbstractOperationHandler
+        implements KeyPresentValidator {
+    public PurchaseOperationHandler() {
+        super(Operation.PURCHASE);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                      FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        validateMap(fruitQuantityMap, transaction);
```

Comment:
The `validateMap` method from `KeyPresentValidator` interface should be implemented and used here to ensure the fruit is present in the `fruitQuantityMap` before processing the transaction.